sqlalchemy_dremio/base.py

```diff
@@ -0,0 +1,266 @@
+                self.process(join.onclause) + ')')
+
+    def visit_extract(self, extract, **kw):
+        field = self.extract_map.get(extract.field, extract.field)
+        return 'DATEPART("%s", %s)' % \
+                    (field, self.process(extract.expr, **kw))
+
+class DremioDDLCompiler(compiler.DDLCompiler):
+    def get_column_specification(self, column, **kwargs):
+        if column.table is None:
+            raise exc.CompileError(
+                            "dremio requires Table-bound columns "
+                            "in order to generate DDL")
+
+        colspec = self.preparer.format_column(column)
+        seq_col = column.table._autoincrement_column
+        if seq_col is column:
+            colspec += " AUTOINCREMENT"
+        else:
+            colspec += " " + self.dialect.type_compiler.process(column.type)
+
+            if column.nullable is not None and not column.primary_key:
+                if not column.nullable or column.primary_key:
+                    colspec += " NOT NULL"
+                else:
+                    colspec += " NULL"
+
+            default = self.get_column_default_string(column)
+            if default is not None:
+                colspec += " DEFAULT " + default
+
+        return colspec
+
+    def visit_drop_index(self, drop):
+        index = drop.element
+        self.append("\nDROP INDEX [%s].[%s]" % \
+                        (index.table.name,
+                        self._index_identifier(index.name)))
+
+class DremioIdentifierPreparer(compiler.IdentifierPreparer):
+    reserved_words = compiler.RESERVED_WORDS.copy()
+    reserved_words.update(['value', 'text'])
+    def __init__(self, dialect):
+        super(DremioIdentifierPreparer, self).\
+                __init__(dialect, initial_quote='[', final_quote=']')
+
+
+
+class DremioDialect(default.DefaultDialect):
+    colspecs = {
+        types.Unicode: AcUnicode,
+        types.Integer: AcInteger,
+        types.SmallInteger: AcSmallInteger,
+        types.Numeric: AcNumeric,
+        types.Float: AcFloat,
+        types.DateTime: AcDateTime,
+        types.Date: AcDate,
+        types.String: AcString,
+        types.LargeBinary: AcBinary,
+        types.Boolean: AcBoolean,
+        types.Text: AcText,
+        types.CHAR: AcChar,
+        types.TIMESTAMP: AcTimeStamp,
+    }
+    name = 'dremio'
+    supports_sane_rowcount = False
+    supports_sane_multi_rowcount = False
+
+    poolclass = pool.SingletonThreadPool
+    statement_compiler = DremioCompiler
+    ddl_compiler = DremioDDLCompiler
+    preparer = DremioIdentifierPreparer
+    execution_ctx_cls = DremioExecutionContext
+
+    @classmethod
+    def dbapi(cls):
+        import pyodbc as module
+        return module
+
+    def create_connect_args(self, url):
+        opts = url.translate_connect_args()
+        connectors = ["Driver={Dremio Driver (*.mdb)}"]
+        connectors.append("Dbq=%s" % opts["database"])
```

Comment:
No concept of Dbq in Dremio either.
Need to add HOST, PORT, ConnectionType, AuthenticationType to the connect_opts/connectors from the Dremio DSN scheme below.
`Driver={};ConnectionType=Direct;HOST={};PORT={};AuthenticationType=Plain;UID={};PWD={}`
Also need to verify that HOST and PORT are appropriately yanked from non-DSN URLs.
Need to check if schema, db, or other terms are supported in the dsn to specify the space and folder.
Trial and error.
Also look here for clues https://github.com/dremio/dremio-oss/blob/master/sabot/kernel/src/test/java/com/dremio/exec/sql/TestInfoSchema.java